service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOControllerTest.java

```diff
@@ -0,0 +1,150 @@
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement;
+import uk.gov.hmcts.reform.fpl.service.DraftCMOService;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(DraftCMOController.class)
+@OverrideAutoConfiguration(enabled = true)
+@SuppressWarnings("unchecked")
+class DraftCMOControllerTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    @Autowired
+    private DraftCMOService draftCMOService;
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper mapper;
+
+    private final LocalDate date = LocalDate.now();
+    private final List<Element<HearingBooking>> hearingDetails = createHearingBookings(date);
+
+    @Test
+    void aboutToStartCallbackShouldPopulateHearingDatesList() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of("hearingDetails", hearingDetails))
+                .build())
+            .build();
+
+        List<String> expected = Arrays.asList(
+            draftCMOService.convertDate(date.plusDays(5)),
+            draftCMOService.convertDate(date.plusDays(2)),
+            draftCMOService.convertDate(date));
+
+        actAndAssert(request, expected);
+    }
+
+    @Test
+    void aboutToSubmitShouldPopulateHiddenHearingDateField() throws Exception {
+        List<Element<HearingBooking>> hearingDetails = createHearingBookings(date);
+
+        DynamicList dynamicHearingDates = draftCMOService.buildDynamicListFromHearingDetails(hearingDetails);
+
+        dynamicHearingDates
+            .setValue(
+                DynamicListElement.builder()
+                    .code(date.plusDays(5).toString())
+                    .label(date.plusDays(5).toString())
+                    .build());
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of("cmoHearingDateList", dynamicHearingDates))
+                .build())
+            .build();
+
+        MvcResult response = makeRequest(request, "about-to-submit");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
```

Comment:
all of this can be a private method that returns caseData, you can use this in all your tests

Reply:
As suggested above, fetching all in one private method will be like creating request object, making request and the fetching data. basically doing more than one thing in a method. Also I think requests  data in various tests could be different. Please suggest.

Reply:
moved into private method